js/ui/map.js

```diff
@@ -249,6 +253,30 @@ class Map extends Camera {
      */
     removeControl(control) {
         control.onRemove(this);
+        if (this._controls[control.constructor.name]) {
+            // delete this._controls[control.constructor.name];
+        }
+        return this;
+    }
+
+
+    /**
+     * Changes position of the control
+     *
+     * @param {string | IControl} control The [`IControl`](#IControl) or the name of the
+     * [`IControl`](#IControl) type you want to move.
+     * @param {string} position on the map to which the control will be moved.
+     * Valid values are `'top-left'`, `'top-right'`, `'bottom-left'`, and `'bottom-right'`.
+     * @returns {Map} `this`
+     * @example
+     * map.setControlPosition('LogoControl', 'top-right');
+     *
+     */
+
+    setControlPosition(control, position) {
+        const _control = typeof control === "string" ? this._controls[control] : control;
+        this.removeControl(_control);
+        this.addControl(_control, position);
```

Comment:
I infer that the raison d'être of this method is to allow users to change the position of the wordmark. Why not offer a constructor option to place the wordmark? 

```js
new mapboxgl.Map({logoPosition: 'top-right'})

// vs

var map = new mapboxgl.Map()
map.setControlPosition('LogoControl', 'top-right');
```